Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -240,7 +240,7 @@ internal class _BSONDecoder: Decoder {
     }
 
     /// Initializes `self` with the given top-level container and options.
-    fileprivate init(referencing container: BSONValue,
+    fileprivate init(referencing container: BSON,
```

Comment:
I realize this is just preserving existing behavior, but given that `decode` only takes in a `Document` is there a reason why we shouldn't just take a `Document` here?

Reply:
opened [SWIFT-646](https://jira.mongodb.org/browse/SWIFT-646)